filebeat/crawler/prospector_log.go

```diff
@@ -120,90 +115,72 @@ func (prospector ProspectorLog) scan(path string, output chan *input.FileEvent)
 		}
 
 		// Check the current info against p.prospectorinfo[file]
-		lastinfo, isKnown := prospector.prospectorList[file]
+		lastinfo, isKnown := p.prospectorList[file]
 
-		oldFile := input.File{
-			FileInfo: lastinfo.Fileinfo,
-		}
+		oldFile := input.NewFile(lastinfo.Fileinfo)
 
 		// Create a new prospector info with the stat info for comparison
-		newInfo := harvester.NewFileStat(newFile.FileInfo, prospector.iteration)
+		newInfo := harvester.NewFileStat(newFile.FileInfo, p.iteration)
+
+		// Init harvester with info
+		h, err := harvester.NewHarvester(
+			p.config, &p.config.Harvester, file, newInfo, p.channel)
+		if err != nil {
+			logp.Err("Error initializing harvester: %v", err)
+			continue
+		}
 
 		// Conditions for starting a new harvester:
 		// - file path hasn't been seen before
 		// - the file's inode or device changed
 		if !isKnown {
-			prospector.checkNewFile(newInfo, file, output)
+			p.checkNewFile(h)
 		} else {
-			newInfo.Continue(&lastinfo)
-			prospector.checkExistingFile(newInfo, &newFile, &oldFile, file, output)
+			h.Stat.Continue(&lastinfo)
```

Comment:
Correct. I think we can move some functionality here to the Harvester also from the prospector side. Now that we use all the variables directly from the harvester it gets much more obvious.